exercises/robot-simulator/example.rs

```diff
@@ -0,0 +1,102 @@
+#[derive(PartialEq, Debug, Clone)]
+pub enum Direction {
+    North,
+    East,
+    South,
+    West,
+}
+
+impl Direction {
+    pub fn previous_clockwise(&self) -> Self {
+        match self {
+            &Direction::North => Direction::West,
+            &Direction::East => Direction::North,
+            &Direction::South => Direction::East,
+            &Direction::West => Direction::South,
+        }
+    }
+
+    pub fn next_clockwise(&self) -> Self {
+        match self {
+            &Direction::North => Direction::East,
+            &Direction::East => Direction::South,
+            &Direction::South => Direction::West,
+            &Direction::West => Direction::North,
+        }
+    }
+}
+
+struct Position {
+    x: i32,
+    y: i32,
+}
+
+impl Position {
+    fn new(x: i32, y: i32) -> Self {
+        Position { x: x, y: y }
+    }
+
+    fn advance(self, direction: &Direction) -> Self {
+        match direction {
+            &Direction::North => Self::new(self.x, self.y + 1),
+            &Direction::South => Self::new(self.x, self.y - 1),
+            &Direction::East => Self::new(self.x + 1, self.y),
+            &Direction::West => Self::new(self.x - 1, self.y),
+        }
+    }
+}
+
+pub struct Robot {
+    position: Position,
+    direction: Direction,
+}
+
+impl Robot {
+    pub fn new(x: i32, y: i32, d: &Direction) -> Self {
+        Robot::build(Position::new(x, y), d.clone())
+    }
+
+    fn build(position: Position, direction: Direction) -> Self {
+        Robot {
+            position: position,
+            direction: direction,
+        }
+    }
+
+    pub fn turn_right(self) -> Self {
```

Comment:
I don't always do this myself (maybe I should if I run into a compiler output I don't understand), but rustc should have outputted ```src/lib.rs:67:21: 67:25 help: run `rustc --explain E0507` to see a detailed explanation```

And I found that one of the solutions  `rustc --explain E0507` does say you can make it `Copy`, so that could have been a way to understand.